### ISSUE TITLE
JSON object is out printed when verification fails on `notation verify`

### ISSUE THREAD

**Issue by shizhMSFT** [notaryproject/notation#546]
### What is the areas you experience the issue in?

Notation CLI

### What is not working as expected?

On verification failure,
```console
$ notation verify <image_ref> --output json
signature verification failed for all the signatures associated with <image_ref>
```
The output is not in the JSON format and there is no failure details. Therefore, scripts cannot further process it.

### What did you expect to happen?

The expected output would be
```console
$ notation verify <image_ref> --output json
{
    "reference": "<image_ref>",
    "reason": "signature verification failed for all the signatures associated with <image_ref>",
    "details": [
        // detailed outcome
    ],
    "result": "Failed"
}
```

Related to https://github.com/notaryproject/notation/pull/527#discussion_r1101444246

**Comment by priteshbandi** [comment]
Moving conversation from PR to here
[priteshbandi](https://github.com/priteshbandi) [Feb 9, 2023](https://github.com/notaryproject/notation/pull/527#discussion_r1102350307) •

`result` can have two values, one is `skip` and other one is `success`. IMO for failure case displaying JSON for wont be useful because it wont contain any useful information which automation/script can use(non-zero exit code should suffice to show failure)

[shizhMSFT](https://github.com/shizhMSFT) [Feb 9, 2023](https://github.com/notaryproject/notation/pull/527#discussion_r1102353528)

The JSON output is usually consumed by scripts or programs. How can them obtain the structured error message and detailed verification outcomes?

[priteshbandi](https://github.com/priteshbandi) [Feb 9, 2023](https://github.com/notaryproject/notation/pull/527#discussion_r1102362533)
To display error message/detailed verification output we need more than `result` field something like `resultReason` because `result` will only say `failure`. Also, in current state it wont be of much help since as there is only one genuine/expected failure use case i.e failed signature verification for all the signatures. Apart from genuine/expected error, all other errors should always be emitted as stdErr.

Are you suggest we should emit json for expected failure use cases or all failure use cases?

**Comment by shizhMSFT** [comment]
> Are you suggest we should emit json for expected failure use cases or all failure use cases?

I am suggesting emitting json for expected failure use cases.

**Comment by byronchien** [comment]
is there a specific use case for consuming the messages programmatically? i feel like message strings like reason and details would be hard to write scripts around. the benefit i could see in keeping them would be that users can log the messages on skipped/failed result status, but notation should already be logging them stderr

**Comment by yizha1** [comment]
Closed by #551
